### PR TITLE
CIV-9201 check your answers page for Trial arrangements

### PR DESCRIPTION
**Pull request by bhagyashreesharma90**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9201

### Change description ###
(It has changes from hearing duration other information page)
Check your answers for finalise trial arrangements
Save data to CCD on submit

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — APPROVED** [review]

File: src/main/routes/features/caseProgression/trialArrangements/checkAnswersController.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+trialCheckAnswersController.post(TRIAL_ARRANGEMENTS_CHECK_YOUR_ANSWERS, (async (req: AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const claim = await getCaseDataFromStore(claimId);
+    const trialReadyCCD: CCDTrialArrangementDefendent = translateDraftTrialArrangementsToCCD(claim);
+    await civilServiceClient.submitDefendantTrialArrangement(req.params.id, trialReadyCCD, req);
+    await deleteDraftClaimFromStore(req.params.id);
+    res.redirect(constructResponseUrlWithIdParams(req.params.id, CP_FINALISE_TRIAL_ARRANGEMENTS_CONFIRMATION_URL));
+  }catch (error) {
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/caseProgression/trialArrangements/checkAnswersController.ts:52]: Space between "}" and "catch" missing.

**Review by m-meulendijks-v1 — DISMISSED** [review]

File: src/main/routes/features/caseProgression/trialArrangements/checkAnswersController.ts (hunk 1 above)
Comment (bhagyashreesharma90) [reply src/main/routes/features/caseProgression/trialArrangements/checkAnswersController.ts]: > Space between "}" and "catch" missing.

done

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

File: src/main/services/features/response/checkAnswers/caseProgression/trialArrangements/buildTrialReadyConfirmationSection.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export const buildIsCaseReadyForTrialOrHearing = (claim: Claim, claimId: string, lang: string | unknown): SummarySection => {
+  let trialReadySummarySections: SummarySection = null;
+
+  trialReadySummarySections = summarySection({
+    title: t('PAGES.CHECK_YOUR_ANSWER.RESPONSE_TITLE', {lng: getLng(lang)}),
+    summaryRows: [],
+  });
+
+  trialReadySummarySections.summaryList.rows.push(summaryRow(t('PAGES.IS_CASE_READY.IS_CASE_READY', { lng: getLng(lang) }), t(`${claim.caseProgression.defendantTrialArrangements.isCaseReady}`, { lng: getLng(lang) }), constructResponseUrlWithIdParams(claimId, IS_CASE_READY_URL), changeLabel(lang)));
+  trialReadySummarySections.summaryList.rows.push(summaryRow(t('PAGES.FINALISE_TRIAL_ARRANGEMENTS.ARE_THERE_ANY_CHANGES', { lng: getLng(lang) }), '<p class="govuk-section-break--visible" >'+t(`${claim.caseProgression.defendantTrialArrangements.hasAnythingChanged.option}`, { lng: getLng(lang) }) +'</p><p>' + t(`${claim.caseProgression.defendantTrialArrangements.hasAnythingChanged.textArea}`, { lng: getLng(lang) })+'</p>', constructResponseUrlWithIdParams(claimId, HAS_ANYTHING_CHANGED_URL), changeLabel(lang)));
```
Comment (LeonardoPalmeiro) [inline src/main/services/features/response/checkAnswers/caseProgression/trialArrangements/buildTrialReadyConfirmationSection.ts:23]: my suggestion in here
  trialReadySummarySections.summaryList.rows.push(summaryRow(t('PAGES.FINALISE_TRIAL_ARRANGEMENTS.ARE_THERE_ANY_CHANGES', { lng: getLng(lang) }), '<p >'+t(`${claim.caseProgression.defendantTrialArrangements.hasAnythingChanged.option}`, { lng: getLng(lang) }) +'</p><hr class="govuk-section-break--visible"><p>' + t(`${claim.caseProgression.defendantTrialArrangements.hasAnythingChanged.textArea}`, { lng: getLng(lang) })+'</p>', constructResponseUrlWithIdParams(claimId, HAS_ANYTHING_CHANGED_URL), changeLabel(lang)));

then we will have the spacing of the divider correctly.
image

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]

File: src/main/services/features/response/checkAnswers/caseProgression/trialArrangements/buildTrialReadyConfirmationSection.ts (hunk 2 above)
Comment (bhagyashreesharma90) [reply src/main/services/features/response/checkAnswers/caseProgression/trialArrangements/buildTrialReadyConfirmationSection.ts]: > my suggestion in here `trialReadySummarySections.summaryList.rows.push(summaryRow(t('PAGES.FINALISE_TRIAL_ARRANGEMENTS.ARE_THERE_ANY_CHANGES', { lng: getLng(lang) }), '<p >'+t(`${claim.caseProgression.defendantTrialArrangements.hasAnythingChanged.option}`, { lng: getLng(lang) }) +'</p><hr class="govuk-section-break--visible"><p>' + t(`${claim.caseProgression.defendantTrialArrangements.hasAnythingChanged.textArea}`, { lng: getLng(lang) })+'</p>', constructResponseUrlWithIdParams(claimId, HAS_ANYTHING_CHANGED_URL), changeLabel(lang)));`
> 
> then we will have the spacing of the divider correctly. image

Done

**Review by kurt-uk — DISMISSED** [review]

**Review by maciejmodzelewski — CHANGES_REQUESTED** [review]
You have some issues in your build
image

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by maciejmodzelewski — APPROVED** [review]

**Pull request closed without merging**